src/Generation/SnippetGenerator.php

```diff
@@ -0,0 +1,441 @@
+
+            // TODO: ensure responsetype/ status classes are ended to imports
+            return AST::if($var->operationSucceeded())
+                ->then(
+                    // Custom operations and google.protobuf.Empty responses have no result.
+                    $isCustomOp || $method->hasEmptyLroResponse
+                        ? $noResult
+                        : Vector::new([
+                            AST::literal("/** @var {$method->lroResponseType->name} {$responseVar->toCode()} */"),
+                            AST::assign($result, $var->getResult()),
+                            AST::call("\0printf")(
+                                AST::literal(
+                                    "'Response data: %s' . PHP_EOL, {$result->toCode()}->serializeToJsonString()"
+                                )
+                            )
+                        ])
+                )->else(
+                    AST::literal("/** @var Status {$error->toCode()} */"),
+                    AST::assign($error, $var->getError()),
+                    AST::call("\0printf")(
+                        AST::literal(
+                            "'Operation failed with data: %s' . PHP_EOL, {$error->toCode()}->serializeToJsonString()"
+                        )
+                    )
+                );
+        };
+        
+        return $this->buildSnippetStructure(
+            $method,
+            $snippetDetails,
+            [
+                AST::literal("/** @var OperationResponse {$responseVar->toCode()} */"),
+                AST::assign(
+                    $responseVar,
+                    AST::call(
+                        $snippetDetails->serviceClientVar,
+                        AST::method($method->methodName)
+                    )($snippetDetails->rpcArguments)
+                ),
+                $responseVar->pollUntilComplete(),
+                PHP_EOL,
+                $useResponseFn($responseVar)
+            ]
+        );
+    }
+
+    private function rpcMethodExamplePaginated(MethodDetails $method, SnippetDetails $snippetDetails): AST
+    {
+        $responseVar = AST::var('response');
+        $page = AST::var('page');
+        $isMap = $method->resourcesField->isMap;
+        $element = AST::var('element');
+        $indexVar = $isMap ? AST::var('key') : null;
+
+        return $this->buildSnippetStructure(
+            $method,
+            $snippetDetails,
+            [
+                '// Iterate over pages of elements',
+                AST::assign(
+                    $responseVar,
+                    AST::call(
+                        $snippetDetails->serviceClientVar,
+                        AST::method($method->methodName)
+                    )($snippetDetails->rpcArguments)
+                ),
+                AST::foreach($responseVar->iteratePages(), $page)(
+                    // TODO: figure out how to get the type of the element being iterated over
```

Comment:
If the method is Paginated, it will contain a `Type $resourceType` ([link](https://github.com/googleapis/gapic-generator-php/blob/15e8620f0abe3adbc76517962b12445edf90f1ce/src/Generation/MethodDetails.php#L205)) which is the repeated resource message field being iterated over.

Don't need to resolve this TODO in this PR if you don't want to, could follow on.

Reply:
I tried that, and it was returning an array type for me - still investigating.